Backpack-SwiftUI/Tests/Calendar/BPKCalendarTests.swift

```diff
@@ -0,0 +1,122 @@
+    
+    func test_singleSelectionCalendar() {
+        let testDate = Calendar.current.date(from: DateComponents(year: 2020, month: 2, day: 5))!
+        
+        assertSnapshot(
+            BPKCalendar(
+                selectionType: .single(
+                    selected: .constant(testDate),
+                    accessibilityConfigurations: SingleAccessibilityConfigurations(
+                        selectionHint: ""
+                    )
+                ),
+                calendar: Calendar.current,
+                validRange: validStart...validEnd
+            )
+            .frame(width: 320, height: 720)
+        )
+    }
+    
+    func test_rangeSelectionCalendar_sameMonth() {
+        let selectionStart = Calendar.current.date(from: DateComponents(year: 2020, month: 2, day: 5))!
+        let selectionEnd = Calendar.current.date(from: DateComponents(year: 2020, month: 2, day: 18))!
+        
+        assertSnapshot(
+            BPKCalendar(
+                selectionType: .range(
+                    selectedRange: .constant(selectionStart...selectionEnd),
+                    accessibilityConfigurations: rangeAccessibilityConfig
+                ),
+                calendar: Calendar.current,
+                validRange: validStart...validEnd
+            )
+            .frame(width: 320, height: 720)
+        )
+    }
+    
+    func test_rangeSelectionCalendar_differentMonth() {
+        let selectionStart = Calendar.current.date(from: DateComponents(year: 2020, month: 1, day: 28))!
+        let selectionEnd = Calendar.current.date(from: DateComponents(year: 2020, month: 2, day: 5))!
+        
+        assertSnapshot(
+            BPKCalendar(
+                selectionType: .range(
+                    selectedRange: .constant(selectionStart...selectionEnd),
+                    accessibilityConfigurations: rangeAccessibilityConfig
+                ),
+                calendar: Calendar.current,
+                validRange: validStart...validEnd
+            )
+            .frame(width: 320, height: 720)
+        )
+    }
+    
+    func test_rangeSelectionCalendar_sameDay() {
+        let selectionStart = Calendar.current.date(from: DateComponents(year: 2020, month: 2, day: 5))!
+        let selectionEnd = Calendar.current.date(from: DateComponents(year: 2020, month: 2, day: 5))!
+        
+        assertSnapshot(
+            BPKCalendar(
+                selectionType: .range(
+                    selectedRange: .constant(selectionStart...selectionEnd),
+                    accessibilityConfigurations: rangeAccessibilityConfig
+                ),
+                calendar: Calendar.current,
+                validRange: validStart...validEnd
+            )
+            .frame(width: 320, height: 720)
+        )
+    }
+    
+    func test_rangeCalendarDayCells() {
+        let calendar = Calendar.current
+        let date = calendar.date(from: .init(year: 2023, month: 11, day: 8))!
+        
+        assertSnapshot(
```

Comment:
👍 very useful to have all the states tested, and visualised through the snapshot 😄